src/Symfony/Component/HttpFoundation/RequestMatcher/RequestMatcher.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\RequestMatcher;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * RequestMatcher compares a pre-defined set of checks against a Request instance.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class RequestMatcher extends ChainRequestMatcher
+{
+    private $host;
+    private $methods;
+    private $ips;
+    private $attributes;
+    private $schemes;
+
+    /**
+     * @param string|null $path
+     * @param string|null $host
+     * @param string[]    $methods
+     * @param string[]    $ips
+     * @param array       $attributes
+     * @param string[]    $schemes
+     */
+    public function __construct($path = null, $host = null, array $methods = array(), array $ips = array(), array $attributes = array(), array $schemes = array())
+    {
+        $matchers = array();
+        if (null !== $path) {
+            $matchers[] = new PathRequestMatcher($path, true);
```

Comment:
In order to make it even more reusable it would be nice to be able to inject custom matchers:
```php
    /**
     * @param RequestMatcherInferace|string|null $path
     * @param RequestMatcherInferace|string|null $host
     * @param RequestMatcherInferace|string[]    $methods
     * @param RequestMatcherInferace|string[]    $ips
     * @param RequestMatcherInferace|array       $attributes
     * @param RequestMatcherInferace|string[]    $schemes
     */
    public function __construct($path = null, $host = null, array $methods = array(), array $ips,  array(), array $attributes = array(), array $schemes = array())
    {
        $matchers = array();
        if (null !== $path) {
            $matchers[] = $path instanceof RequestMatcherInterface ? $path : new PathRequestMatcher($path, true);
        }

        // ...
 ```